src/session-pool.ts

```diff
@@ -217,6 +217,36 @@ export function isSessionNotFoundError(
   );
 }
 
+/**
+ * Checks whether the given error is a 'Database not found' error.
+ * @param error the error to check
+ * @return true if the error is a 'Database not found' error, and otherwise false.
+ */
+export function isDatabaseNotFoundError(
+  error: grpc.ServiceError | undefined
+): boolean {
+  return (
+    error !== undefined &&
+    error.code === grpc.status.NOT_FOUND &&
+    error.message.includes('Database not found')
+  );
+}
+
+/**
+ * Checks whether the given error is an 'Instance not found' error.
+ * @param error the error to check
+ * @return true if the error is an 'Instance not found' error, and otherwise false.
```

Comment:
```suggestion
 * @param {Error} error The error to check.
 * @return {boolean} True if the error is an 'Instance not found' error, and otherwise false.
```